src/rust/src/lib.rs

```diff
@@ -107,6 +107,12 @@ pub use fallible_streaming_iterator::FallibleStreamingIterator;
 pub use geozero::{FeatureAccess, FeatureProperties, GeozeroGeometry};
 
 pub const VERSION: u8 = 3;
-pub const MAGIC_BYTES: [u8; 8] = [b'f', b'g', b'b', VERSION, b'f', b'g', b'b', 0];
+const MAGIC_BYTES: [u8; 8] = [b'f', b'g', b'b', VERSION, b'f', b'g', b'b', 0];
 
-pub const HEADER_MAX_BUFFER_SIZE: usize = 1048576 * 10;
+const HEADER_MAX_BUFFER_SIZE: usize = 1048576 * 10;
+
+fn check_magic_bytes(magic_bytes: &[u8]) -> bool {
+    magic_bytes[0..3] == MAGIC_BYTES[0..3]
+        && magic_bytes[4..7] == MAGIC_BYTES[4..7]
```

Comment:
```suggestion
        && magic_bytes[4..8] == MAGIC_BYTES[4..8]
```

Do we need to veritfy the final `0` as well?

Reply:
I guess you're right. Despite being added just for padding, the trailing zero is part of the magic bytes.